File: e2e/testdata/node/features/network_test.js
const http = require('http');
const dns = require('dns');
const net = require('net');

console.log('Testing network access...');

// 测试HTTP访问
const req = http.get('http://httpbin.org/get', (res) => {
    console.log('HTTP access succeeded - this should be blocked');
    res.on('data', () => {});
}).on('error', (e) => {
    console.log(`Network access failed: ${e.message}`);
});
req.setTimeout(3000, () => {
    req.destroy();
    console.log('HTTP request timed out (expected)');
});

// 测试DNS解析
dns.lookup('google.com', (err, address) => {
    if (err) {
        console.log(`DNS resolution failed: ${err.message}`);
    } else {
        console.log('DNS resolution succeeded - this should be blocked');
    }
});

// 测试原始套接字连接
const socket = new net.Socket();
socket.setTimeout(3000);
socket.connect(53, '8.8.8.8', () => {
    console.log('Raw socket connection succeeded - this should be blocked');
    socket.destroy();
}).on('error', (e) => {
    console.log(`Raw socket connection failed: ${e.message}`);
});

setTimeout(() => {
    console.log('Network test completed - network access should be blocked in sandbox');
}, 4000);

File: e2e/testdata/python/features/network_test.py
import socket
import urllib.request
import sys

print("Testing network access...")

# 测试HTTP访问
try:
    response = urllib.request.urlopen('http://httpbin.org/get', timeout=3)
    print("HTTP access succeeded - this should be blocked")
except Exception as e:
    print(f"Network access failed: {type(e).__name__}: {e}")

# 测试DNS解析
try:
    socket.gethostbyname('google.com')
    print("DNS resolution succeeded - this should be blocked")
except Exception as e:
    print(f"DNS resolution failed: {type(e).__name__}: {e}")

# 测试原始套接字连接
try:
    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    sock.settimeout(3)
    sock.connect(('8.8.8.8', 53))
    print("Raw socket connection succeeded - this should be blocked")
    sock.close()
except Exception as e:
    print(f"Raw socket connection failed: {type(e).__name__}: {e}")

print("Network test completed - network access should be blocked in sandbox")